### PR TITLE
INT-2276 - Do not hard fail when some devices cannot be fetched from the computer details API

### PR DESCRIPTION
**Pull request by austinkelleher**

### CODE REVIEW

File: src/steps/devices/index.ts [hunk 1]
```diff
@@ -99,14 +99,15 @@ async function iterateComputerDetails(
...
+      logger.warn(
         {
           err,
           computerId: computer.id,
         },
         'Could not fetch computer by id',
       );
-      numComputerDetailFetchFailed++;
+
+      computerIdsFetchFailedSet.add(computer.id.toString());
```
Comment (aiwilliams) [inline src/steps/devices/index.ts:110]: Any chance the `computer.id` could be `undefined`? I realize the type may say no but you know how these things can go. 😬 

```js
> String(undefined)
'undefined'
> undefined.toString()
Uncaught TypeError: Cannot read property 'toString' of undefined
> undefined?.toString()
undefined
```

File: src/util/device.ts [hunk 2]
```diff
@@ -37,7 +37,7 @@ export async function getComputerDeviceIdToGraphObjectKeyMap(
     throw new IntegrationError({
       message:
         'Could not find computer device ID to graph object _key map in job state',
-      code: 'ACCOUNT_ENTITY_NOT_FOUND',
+      code: 'COMPUTER_DEVICE_ID_TO_GRAPH_OBJECT_KEY_MAP',
```
Comment (aiwilliams) [inline src/util/device.ts:40]: 💯

**Review by aiwilliams — APPROVED** [review]

File: src/steps/devices/index.ts [hunk 3]
```diff
@@ -116,16 +117,21 @@ async function iterateComputerDetails(
...
 
-  if (numComputerDetailFetchFailed) {
-    throw new IntegrationError({
-      message: `Unable to fetch all computer details (success=${numComputerDetailFetchSuccess}, failed=${numComputerDetailFetchFailed})`,
-      code: 'ERROR_FETCH_COMPUTER_DETAILS',
-    });
+  if (computerIdsFetchFailedSet.size) {
+    // The Jamf API is failing intermittemently with 500 errors. The same
+    // computer IDs are causing failures across integration executions. We'll
+    // be reporting this issue to Jamf.
```
Comment (adam-in-ict) [inline src/steps/devices/index.ts:128]: For these computers getting 500 errors, what will the outcome be in J1?  Will it keep stale data on them or will they be flagged for deletion when they no longer show up in the collected data for a given instance of this integration?

**Review by ceelias — APPROVED** [review]

File: src/steps/devices/index.ts (hunk 3 above)
Comment (austinkelleher) [reply src/steps/devices/index.ts:133]: The worst case is that the entities that receive 500 errors will be deleted from the platform. It seems that the same entities are having the issue, so they've probably never been ingested in the first place.

**Review by adam-in-ict — APPROVED** [review]

File: src/steps/devices/index.ts (hunk 3 above)
Comment (adam-in-ict) [reply src/steps/devices/index.ts:133]: I was primarily curious if they'd be able to easily query J1 to see which computers all of a sudden lost detail or if those computers would disappear entirely.  Thanks!

**Pull request merged**